src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -278,4 +278,15 @@ protected Object access(VirtualFrame frame, DynamicObject receiver, String name,
 
     }
 
+    @Resolve(message = "NEW")
+    public static abstract class ForeignNewNode extends Node {
+
+        @Child private DispatchHeadNode dispatchHeadNode = insert(new DispatchHeadNode(true, false, MissingBehavior.CALL_METHOD_MISSING, DispatchAction.CALL_METHOD));
```

Comment:
No need for `insert`.
Could you use `CallDispatchHeadNode.createOnSelf()` which is shorter and hard-codes less?
Actually I'm not sure we should ignore visibility here, `new` is normally public, so `CallDispatchHeadNode.create()` would be best.